apps/server/src/services/category.service.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  CreateCategoryInput,
+  UpdateCategoryInput,
+} from './../schemas/category.schema';
+import { ICategoryService } from './../interface/category/ICategoryService';
+import { CategoryRepository } from '../repositories/category.repository';
+
+import { NotFoundError } from '../utils/error';
+import { Category } from '@bitafish/shared-types';
+
+export class CategoryService implements ICategoryService {
+  private repository: CategoryRepository;
+
+  constructor(_repository: CategoryRepository) {
+    this.repository = _repository;
+  }
+
+  async getCategories(limit: number, offset: number) {
+    return await this.repository.find(limit, offset);
+  }
+  async createCategory({ name }: CreateCategoryInput) {
+    try {
+      const category = await this.repository.create({
+        name,
+      });
+      return category;
```

Comment:
here you have to check if there's already a category exists with this name **if found** you should return an error saying that category with this name already exists **if not ** then you create the category